src/genkey/clu_genkey_setup.c

```diff
@@ -124,48 +125,39 @@ int wolfCLU_genKeySetup(int argc, char** argv)
         if (ret > 0) {
             if (argv[ret+1] != NULL) {
                 if (XSTRNCMP(argv[ret+1], "pub", 3) == 0)
-                    directiveArg = PUB_ONLY;
+                    directiveArg = PUB_ONLY_FILE;
                 else if (XSTRNCMP(argv[ret+1], "priv", 4) == 0)
-                    directiveArg = PRIV_ONLY;
+                    directiveArg = PRIV_ONLY_FILE;
                 else if (XSTRNCMP(argv[ret+1], "keypair", 7) == 0)
                     directiveArg = PRIV_AND_PUB;
             }
-        } else {
+        }
+        else {
             printf("No -output <PUB/PRIV/KEYPAIR>\n");
             printf("DEFAULT: output public and private key pair\n");
             directiveArg = PRIV_AND_PUB;
         }
 
-        /* get the size argument */
-        ret = wolfCLU_checkForArg("-size", 5, argc, argv);
+        /* get the curve name */
+        ret = wolfCLU_checkForArg("-name", 5, argc, argv);
         if (ret > 0) {
             if (argv[ret+1] != NULL) {
-                char* cur;
-                /* make sure it's an integer */
-                if (*argv[ret+1] == '\0') {
-                    printf("Empty -size argument, using 32\n");
-                    sizeArg = 32;
-                }
-                else {
-                    for (cur = argv[ret+1]; *cur && isdigit(*cur); ++cur);
-                    if (*cur == '\0') {
-                        sizeArg = atoi(argv[ret+1]);
-                    }
-                    else {
-                        printf("Invalid -size (%s), using 32\n",
-                               argv[ret+1]);
-                        sizeArg = 32;
-                    }
-                }
+                int i;
+
+                name = argv[ret+1];
+
+                /* convert name to upper case */
+                for (i = 0; i < XSTRLEN(name); i++)
+                    (void)toupper(name[i]);
             }
-        } else {
-            printf("No -size <SIZE>\n");
+        }
+
+        if (name == NULL) { /* if we have the name of curve we know */
             printf("DEFAULT: use a 32 ECC key\n");
-            sizeArg = 32;
         }
 
         ret = wolfCLU_genKey_ECC(&rng, keyOutFName, directiveArg,
-                                 formatArg, sizeArg);
+                                 formatArg, NULL);
```

Comment:
Should this be `name` instead of `NULL` for the last param? As it stands, it doesn't look like `name` is ever used.